core/src/main/java/nl/tudelft/contextproject/core/input/KeyboardInputProcessor.java

```diff
@@ -88,9 +108,37 @@ public void update(float dt) {
         }
 
         if (isToggled()){
-            player.getColourPalette().cycle();
+            players.get(activePlayer).getColourPalette().cycle();
             toggled = false;
         }
+
+        if (toggle1){
```

Comment:
Method is about 50 lines long, this violates check-style. you can rewrite this, maybe with a boolean[] and a loop which should map nicely on to the list of players we have anyways.